tests/trainer/test_states.py

```diff
@@ -0,0 +1,132 @@
+from pytorch_lightning import Trainer, Callback
+from pytorch_lightning.trainer.states import TrainerState
+from tests.base import EvalModelTemplate
+
+
+def test_initialize_state(tmpdir):
+    """
+    Tests that state is INITIALIZE after Trainer creation
+    """
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        fast_dev_run=True,
+    )
+
+    assert trainer.state == TrainerState.INITIALIZE
+
+
+def test_running_state_during_fit(tmpdir):
+    """
+        Tests that state is set to RUNNING during fit
+    """
+
+    class StateSnapshotCallback(Callback):
```

Comment:
can we define this outside the function? we don't want to duplicate code. 
Also I would rather test separately both max_steps=1 and fast_dev_run (you can parameterize the test with these args)

Reply:
Done